enumerable.rb

```diff
@@ -0,0 +1,99 @@
+module Enumerable
+  def my_each(array)
+    array.length.times { |index| yield(array[index]) }
+
+    nil
+  end
+
+  def my_each_with_index(array)
+    0.upto(array.length - 1) { |index| yield(array[index], index) }
+
+    nil
+  end
+
+  def my_select(array)
+    result = []
+
+    my_each(array) { |item| result << item if yield(item) }
+
+    result
+  end
+
+  def my_all?(array)
+    result = true
+
+    my_each(array) { |item| result = false unless yield(item) }
+
+    result
+  end
+
+  def my_any?(array)
+    result = false
+
+    my_each(array) { |item| result = true if yield(item) }
+
+    result
+  end
+
+  def my_none?(array)
+    result = true
+
+    my_each(array) { |item| result = false if yield(item) }
+
+    result
+  end
+
+  def my_count(array, value = nil)
+    counter = 0
+
+    my_each(array) do |item|
+      if block_given?
+        counter += 1 if yield(item)
+      elsif item == value
+        counter += 1
+      else
+        counter = array.length unless value
+      end
+    end
+    counter
+  end
+
+  def my_map(array)
+    result = []
+
+    my_each(array) { |item| result << yield(item) }
+
+    result
+  end
```

Comment:
* [ ] should have working `my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

#### my_map 

- [ ] given a block returns a new array with the results of running the block once for every element in enum.
```ruby
[2,4,7,11].my_map {|num| num < 10 } == [2,4,7,11].map {|num| num < 10 
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/77697412-70592000-6f74-11ea-80b1-c9ff3f5bfa63.png)
- Real map
![image](https://user-images.githubusercontent.com/52209067/77697467-85ce4a00-6f74-11ea-9650-70790cae2000.png)

- [ ] returns an Enumerator if no block is given
```ruby
[2,4,7,11].my_map should be equal to [2,4,7,11].map #<Enumerator: [2, 4, 7, 11]:map
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/77697596-c4640480-6f74-11ea-9488-1b55b82f2aa4.png)
- Real map
![image](https://user-images.githubusercontent.com/52209067/77697646-dc3b8880-6f74-11ea-9277-4ed4a2b3af4c.png)

- [ ] in the specifications of the project in the [Odin project](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks) # 12, you were asked to modify your my_map method to take either a proc or a block. And if both a proc and a block are given, only execute the proc. The real map doesn't have this feature, but you were asked to have it so let's work on it

Given 
```ruby
my_proc = Proc.new {|num| num > 10 }
```
Result
```ruby
puts [18, 22, 5, 6] .my_map(my_proc) {|num| num < 10 }  => true true false false
```